src/DevBetterWeb.Infrastructure/PaymentHandler/StripePaymentHandler/StripePaymentHandlerSubscriptionService.cs

```diff
@@ -97,6 +99,31 @@ public string GetStatus(string subscriptionId)
       return status;
     }
 
+    public BillingPeriod GetBillingPeriod(string subscriptionId)
+    {
+      var subscription = GetSubscription(subscriptionId);
+
+      var billingPeriod = GetSubscriptionBillingInterval(subscription);
+
+      return billingPeriod;
+    }
+
+    private BillingPeriod GetSubscriptionBillingInterval(Subscription subscription)
+    {
+      var item = subscription.Items.Data[0];
+      var period = item.Price.Recurring.Interval;
+
+      if (period == "month")
```

Comment:
I think BillingPeriod is a SmartEnum, right?

If so, you can overload its constructor so you don't need this if logic:

```
public BillingPeriod(string stripePeriod)
{
  // put if logic here
 // put new InvalidBillingPeriodException here, too
}
```

Reply:
I had it as a SmartEnum but then I was having trouble putting it in the database and didn't see any real benefit to not just using a regular enum so I switched it to that. I could switch it back to SmartEnum if you think it's worth it for this